PhysicsTools/TensorFlow/interface/TensorFlow.h

```diff
@@ -0,0 +1,81 @@
+
+#include "tensorflow/core/public/session.h"
+#include "tensorflow/core/framework/tensor.h"
+#include "tensorflow/cc/saved_model/loader.h"
+#include "tensorflow/cc/saved_model/tag_constants.h"
+#include "tensorflow/cc/saved_model/constants.h"
+#include "tensorflow/core/lib/io/path.h"
+#include "tensorflow/core/util/tensor_bundle/naming.h"
+
+#include "FWCore/Utilities/interface/Exception.h"
+
+namespace tensorflow
+{
+
+typedef std::pair<std::string, Tensor> NamedTensor;
+typedef std::vector<NamedTensor> NamedTensorList;
+
+// set the tensorflow log level
+void setLogging(const std::string& level = "3");
+
+// updates the config of sessionOptions so that it uses a single thread
+void setThreading(SessionOptions& sessionOptions, int nThreads);
+
+// loads a meta graph definition saved at exportDir using the SavedModel interface for a tag
+// transfers ownership
+MetaGraphDef* loadMetaGraph(const std::string& exportDir, bool multiThreaded = false,
+    const std::string& tag = kSavedModelTagServe);
+
+// return a new, empty session
+// transfers ownership
+Session* createSession(bool multiThreaded = false);
+
+// return a new session that contains an already loaded meta graph whose exportDir must be given in
+// order to load and initialize the variables within the session
+// transfers ownership
+Session* createSession(MetaGraphDef* metaGraph, const std::string& exportDir,
+    bool multiThreaded = false);
+
+// closes a session, calls its destructor, resets the pointer, and returns true on success
+bool closeSession(Session*& session);
+
+// run the session with inputs, outputNames and targetNodes, and store output tensors
+// throws a cms exception when not successful
+void run(Session* session, const NamedTensorList& inputs,
+    const std::vector<std::string>& outputNames, const std::vector<std::string>& targetNodes,
+    std::vector<Tensor>* outputs);
+
+// run the session with inputNames, inputTensors, outputNames and targetNodes, and store output
+// tensors
+// throws a cms exception when not successful
+void run(Session* session, const std::vector<std::string>& inputNames,
+    const std::vector<Tensor>& inputTensors, const std::vector<std::string>& outputNames,
+    const std::vector<std::string>& targetNodes, std::vector<Tensor>* outputs);
+
+// run the session with inputs and outputNames, and store output tensors
+// throws a cms exception when not successful
+void run(Session* session, const NamedTensorList& inputs,
+    const std::vector<std::string>& outputNames, std::vector<Tensor>* outputs);
+
+// run the session with inputNames, inputTensors and outputNames, and store output tensors
+// throws a cms exception when not successful
+void run(Session* session, const std::vector<std::string>& inputNames,
+    const std::vector<Tensor>& inputTensors, const std::vector<std::string>& outputNames,
+    std::vector<Tensor>* outputs);
+
+} // namespace tensorflow
+
+// use a namespace alias
+namespace tf = tensorflow;
```

Comment:
it's unclear if this aliasing is really needed.
Better use what's in the upstream (```tensorflow```).